File: android/src/main/java/org/ligi/gobandroid_hd/ui/share/ShareSGFDialog.java
package org.ligi.gobandroid_hd.ui.share;

import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.Uri;
import android.widget.RadioGroup;
import butterknife.BindView;
import butterknife.ButterKnife;
import java.io.File;
import org.ligi.axt.listeners.DialogDiscardingOnClickListener;
import org.ligi.gobandroid_hd.R;
import org.ligi.gobandroid_hd.ui.GoBoardViewHD;
import org.ligi.gobandroid_hd.ui.GobandroidDialog;

/**
 * Dialog with the intention to share the current Game
 *
 * @author <a href="http://ligi.de">Marcus -Ligi- Bueschleb</a>
 *         <p/>
 *         License: This software is licensed with GPLv3
 */
public class ShareSGFDialog extends GobandroidDialog {

    @BindView(R.id.shareTypeRadioGroup)
    RadioGroup typeSpinner;

    public ShareSGFDialog(Context context) {
        super(context);

        setContentView(R.layout.share_options);
        ButterKnife.bind(this, this);

        setTitle(R.string.share);
        setIconResource(R.drawable.ic_social_share);

        setNegativeButton(R.string.cancel, new DialogDiscardingOnClickListener());
        setPositiveButton(R.string.ok, new OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                switch (typeSpinner.getCheckedRadioButtonId()) {
                    case R.id.radioButtonAsUnicode:
                        final Intent intent = new Intent(Intent.ACTION_SEND);
                        intent.putExtra(Intent.EXTRA_TEXT, gameProvider.get().getVisualBoard().toString(true));
                        intent.setType("text/plain");
                        getContext().startActivity(Intent.createChooser(intent, getContext().getString(R.string.choose_how_to_send_sgf)));
                        break;

                    case R.id.radioButtonAsAttachment:
                        new ShareAsAttachmentDialog(getContext());
                        break;

                    case R.id.radioButtonAsImage:
                        final File file = new File(settings.getSGFBasePath(), "game_to_share_via_action.png");

                        final GoBoardViewHD goBoardViewHD = new GoBoardViewHD(getContext());
                        final Bitmap mutableImage = BitmapFactory.decodeResource(getContext().getResources(), R.drawable.shinkaya)
                                                                 .copy(Bitmap.Config.ARGB_8888, true);

                        goBoardViewHD.layout(0, 0, mutableImage.getWidth(), mutableImage.getHeight());
                        goBoardViewHD.screenshot(file, mutableImage);

                        final Intent it = new Intent(Intent.ACTION_SEND);
                        it.putExtra(Intent.EXTRA_SUBJECT, "Image created with gobandroid");
                        it.putExtra(Intent.EXTRA_STREAM, Uri.parse("file://" + file));
                        it.setType("image/*");
                        getContext().startActivity(Intent.createChooser(it, getContext().getString(R.string.choose_how_to_send_sgf)));

                        break;
                }

                dismiss();

            }
        });

    }
}

File: android/src/main/java/org/ligi/gobandroid_hd/ui/share/ShareSGFDialog.kt
package org.ligi.gobandroid_hd.ui.share

import android.content.Context
import android.content.Intent
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import kotlinx.android.synthetic.main.share_options.view.*
import org.ligi.gobandroid_hd.R
import org.ligi.gobandroid_hd.ui.GoBoardView
import org.ligi.gobandroid_hd.ui.GobandroidDialog
import java.io.File
/**
 * Dialog with the intention to share the current Game

 * @author [Marcus -Ligi- Bueschleb](http://ligi.de)
 * *
 *
 *
 * *         License: This software is licensed with GPLv3
 */
class ShareSGFDialog(context: Context) : GobandroidDialog(context) {

    init {

        setContentView(R.layout.share_options)

        setTitle(R.string.share)
        setIconResource(R.drawable.ic_social_share)

        setNegativeButton(R.string.cancel)
        setPositiveButton(R.string.ok,  { _ ->
            when (container.shareTypeRadioGroup.checkedRadioButtonId) {
                R.id.radioButtonAsUnicode -> {
                    val intent = Intent(Intent.ACTION_SEND)
                    intent.putExtra(Intent.EXTRA_TEXT, gameProvider.get().visualBoard.toString(true))
                    intent.type = "text/plain"
                    getContext().startActivity(Intent.createChooser(intent, getContext().getString(R.string.choose_how_to_send_sgf)))
                }

                R.id.radioButtonAsAttachment -> ShareAsAttachmentDialog(getContext())

                R.id.radioButtonAsImage -> {
                    val file = File(settings.SGFBasePath, "game_to_share_via_action.png")

                    val goBoardView = GoBoardView(getContext())
                    val mutableImage = BitmapFactory.decodeResource(getContext().resources, R.drawable.shinkaya)
                            .copy(Bitmap.Config.ARGB_8888, true)

                    goBoardView.layout(0, 0, mutableImage.width, mutableImage.height)
                    goBoardView.screenshot(file, mutableImage)

                    val it = Intent(Intent.ACTION_SEND)
                    it.putExtra(Intent.EXTRA_SUBJECT, "Image created with gobandroid")
                    it.putExtra(Intent.EXTRA_STREAM, Uri.parse("file://" + file))
                    it.type = "image/*"
                    getContext().startActivity(Intent.createChooser(it, getContext().getString(R.string.choose_how_to_send_sgf)))
                }
            }

            dismiss()
        })

    }
}
